src/fhir.js

```diff
@@ -301,15 +391,76 @@ class Patient extends FHIRObject {
       return [];
     }
     const resourceType = classInfo.name.replace(/^FHIR\./, '');
-    const records = this._bundle.entry.filter( e => {
-      return e.resource && e.resource.resourceType == resourceType;
-    }).map( e => {
-      return new FHIRObject(e.resource, classInfo, this._modelInfo);
-    });
+    const records = this._bundle.entry
+      .filter(e => {
+        return e.resource && e.resource.resourceType == resourceType;
+      })
+      .map(e => {
+        return new FHIRObject(e.resource, classInfo, this._modelInfo);
+      });
     return records;
   }
 }
 
+class AsyncPatient extends FHIRObject {
+  constructor(patientData, modelInfo, fhirClient) {
+    const patientClass = modelInfo.patientClassIdentifier
+      ? modelInfo.patientClassIdentifier
+      : modelInfo.patientClassName;
+    const ptClass = modelInfo.findClass(patientClass);
+    super(patientData, ptClass, modelInfo);
+
+    // Define a "private" un-enumerable property to hold the patient data
+    Object.defineProperty(this, '_patientData', {
+      value: patientData,
+      enumerable: false
+    });
+    this.fhirClient = fhirClient;
+  }
+
+  async findRecords(profile) {
+    const classInfo = this._modelInfo.findClass(profile);
+    if (classInfo == null) {
+      console.error(`Failed to find type info for ${profile}`);
+      return [];
+    }
+    const resourceType = classInfo.name.replace(/^FHIR\./, '');
+    // If the patient resource type is requested, return array with just this resource
+    if (resourceType === 'Patient') {
+      return [this];
+    }
+
+    const compartmentInfo = patientCompartmentDefinition.resource.filter(
+      def => def.code === resourceType
+    );
+    if (!compartmentInfo[0]) {
+      if (response.data.total > 0) {
```

Comment:
I'd suggest adding a test case for buddy over here. You might surprised what happens ;)